test/SparqlAntlrParserTest.cpp

```diff
@@ -191,460 +222,213 @@ TEST(SparqlParser, ComplexConstructQuery) {
-
-  EXPECT_THAT(triples[10], ElementsAre(IsIri(something),  //
-                                       IsIri(type),       //
-                                       IsIri(somethingElse)));
-}
-
-TEST(SparqlParser, GraphTermNumericLiteral) {
-  string input = "1337";
-  ParserAndVisitor p{input};
 
-  auto graphTerm = p.parser_.graphTerm()->accept(&p.visitor_).as<GraphTerm>();
-  EXPECT_THAT(graphTerm, IsLiteral("1337"));
-}
-
-TEST(SparqlParser, GraphTermBooleanLiteral) {
-  string input = "true";
-  ParserAndVisitor p{input};
-
-  auto graphTerm = p.parser_.graphTerm()->accept(&p.visitor_).as<GraphTerm>();
-  EXPECT_THAT(graphTerm, IsLiteral(input));
-}
-
-TEST(SparqlParser, GraphTermBlankNode) {
-  string input = "[]";
-  ParserAndVisitor p{input};
-
-  auto graphTerm = p.parser_.graphTerm()->accept(&p.visitor_).as<GraphTerm>();
-  EXPECT_THAT(graphTerm, IsBlankNode(true, "0"));
-}
-
-TEST(SparqlParser, GraphTermIri) {
-  string input = "<http://dummy-iri.com#fragment>";
-  ParserAndVisitor p{input};
-
-  auto graphTerm = p.parser_.graphTerm()->accept(&p.visitor_).as<GraphTerm>();
-  EXPECT_THAT(graphTerm, IsIri(input));
-}
-
-TEST(SparqlParser, GraphTermRdfLiteral) {
-  string input = "\"abc\"";
-  ParserAndVisitor p{input};
-
-  auto graphTerm = p.parser_.graphTerm()->accept(&p.visitor_).as<GraphTerm>();
-  EXPECT_THAT(graphTerm, IsLiteral(input));
-}
-
-TEST(SparqlParser, GraphTermRdfNil) {
-  string input = "()";
-  ParserAndVisitor p{input};
-
-  auto graphTerm = p.parser_.graphTerm()->accept(&p.visitor_).as<GraphTerm>();
-  EXPECT_THAT(graphTerm, IsIri(nil));
+  expectCompleteArrayParse(
+      parseConstructQuery(input),
+      ElementsAre(IsBlankNode(true, "0"), IsVariableVariant("?a"),
+                  IsBlankNode(true, "3")),
+      ElementsAre(IsBlankNode(true, "1"), IsIri(first), IsBlankNode(true, "2")),
+      ElementsAre(IsBlankNode(true, "1"), IsIri(rest), IsIri(nil)),
+      ElementsAre(IsBlankNode(true, "2"), IsIri(first),
+                  IsVariableVariant("?c")),
+      ElementsAre(IsBlankNode(true, "2"), IsIri(rest), IsIri(nil)),
+      ElementsAre(IsBlankNode(true, "3"), IsIri(first),
+                  IsVariableVariant("?b")),
+      ElementsAre(IsBlankNode(true, "3"), IsIri(rest), IsBlankNode(true, "1")),
+      ElementsAre(IsBlankNode(true, "0"), IsVariableVariant("?d"),
+                  IsBlankNode(true, "4")),
+      ElementsAre(IsBlankNode(true, "4"), IsVariableVariant("?e"),
+                  IsBlankNode(true, "5")),
+      ElementsAre(IsBlankNode(true, "5"), IsVariableVariant("?f"),
+                  IsVariableVariant("?g")),
+      ElementsAre(IsIri("<http://wallscope.co.uk/resource/olympics/medal/"
+                        "#something>"),
+                  IsIri(type),
+                  IsIri("<http://wallscope.co.uk/resource/olympics/medal/"
+                        "#somethingelse>")));
+}
+
+TEST(SparqlParser, GraphTerm) {
+  auto expectGraphTerm = makeExpectCompleteParser<parseGraphTerm>();
```

Comment:
As I understand it the matchers task is to take a value that has been computed somewhere else and check that it is correct. In your suggestion the matcher would first have to parse the given argument and then actually match it. I think that it might work but we should think about whether this is really good or just mixes responsibilities. But with proper naming e.g. a `ParsedAsEquals<...>` matcher this might work out.

Reply:
I was able to add the source location with 577eadca00ba6d244fa7315f393951f0b5f02a2e and it even seems to be the official, intended way. [Docs](https://github.com/google/googletest/blob/main/docs/advanced.md#using-assertions-in-sub-routines)